include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -0,0 +1,585 @@
+            // aligned sequences which is more efficient if possible.
+            size_t off_end{seq.size() - offset};
+            for (auto chr : get<1>(align))
+                if (chr == gap::GAP)
+                    ++off_end;
+            off_end -= (get<1>(align)).size();
+
+            write_field(stream_it,
+                        detail::get_cigar_string(std::forward<align_type>(align),
+                                                 std::forward<offset_type>(offset),
+                                                 off_end),
+                        separator);
+        }
+        else
+        {
+            stream << "*\t";
+        }
+
+        write_field(stream_it, get<0>(std::forward<mate_type>(mate)), separator);
+
+        stream << get<1>(std::forward<mate_type>(mate)) << separator;
+
+        stream << get<2>(std::forward<mate_type>(mate)) << separator;
+
+        write_field(stream_it, std::forward<seq_type>(seq), separator);
+
+        write_field(stream_it, std::forward<qual_type>(qual));
+
+        write_tag_fields(stream, std::forward<tag_dict_type>(tag_dict), separator);
+
+        detail::write_eol(stream_it, options.add_carriage_return);
+    }
+
+protected:
+    //!/brief The format version string.
+    static constexpr char format_version[4] = "1.6";
+
+    //!\brief A variable that tracks whether the header_ptr as been written or not.
+    bool written_header_ptr{false};
+
+    /*!\brief Writes a field value to the stream adding a separator character.
+     * \tparam stream_it_t The stream iterator type.
+     * \tparam field_type  The type of the field value. Must model std::ranges::ForwardRange.
+     *
+     * \param[in,out] stream_it   The stream iterator to print to.
+     * \param[in]     field_value The value to print.
+     * \param[in]     separator   The field separator to append.
+     */
+    template <typename stream_it_t, typename field_type>
+        requires std::ranges::ForwardRange<field_type>
+    void write_field(stream_it_t & stream_it, field_type && field_value, char const separator)
+    {
+        write_field(stream_it, std::forward<field_type>(field_value));
+        stream_it = separator;
+    }
+
+
+    /*!\brief Writes a field value to the stream.
+     * \tparam stream_it_t The stream iterator type.
+     * \tparam field_type  The type of the field value. Must model std::ranges::ForwardRange.
+     *
+     * \param[in,out] stream_it   The stream iterator to print to.
+     * \param[in]     field_value The value to print.
+     */
+     template <typename stream_it_t, typename field_type>
+        requires std::ranges::ForwardRange<field_type>
+    void write_field(stream_it_t & stream_it, field_type && field_value)
+    {
+        if (ranges::empty(field_value))
+            stream_it = '*';
+        else
+            ranges::copy(field_value | view::to_char
+                                     | view::take_until(is_space)
+                         , stream_it);
```

Comment:
just write this into one line maybe? If it doesnt fit, correct the indention/spacing and enclose the block in braces

Reply:
I would also rename this to `write_range` maybe, because it can't write other fields. And it is also only called once and it is confusing that the above is called the same, but does something different semantically (usually if you omit one parameter, it means that this is defaulted, but here it means that it is not written; thats inconsistent).

Maybe just remove the first interface and print the separator locally.